pinot-controller/src/main/java/org/apache/pinot/controller/api/upload/ZKOperator.java

```diff
@@ -81,18 +88,20 @@ public void completeSegmentOperations(String tableNameWithType, SegmentMetadata
 
     LOGGER.info("Segment {} from table {} already exists, refreshing if necessary", segmentName, tableNameWithType);
     processExistingSegment(segmentMetadata, finalSegmentLocationURI, currentSegmentLocation,
-        enableParallelPushProtection, headers, zkDownloadURI, crypter, tableNameWithType, segmentName,
+        enableParallelPushProtection, replaceOnly, headers, zkDownloadURI, crypter, tableNameWithType, segmentName,
```

Comment:
Why do we need to pass in the boolean here? This method always processes existing segment